backend/server.js

```diff
@@ -63,6 +63,44 @@ app.post('/login', async (req, res) => {
     }
 });
 
+app.post('/activities',async (req,res)=>{
+
+    const {title,description,start_date,end_date}=req.body;
+    if(!title || !description || !start_date || !end_date)
+    {
+        return res.status(400).json('All fields are required');
+    }
+    
+    try{
+    
+        if(!req.auth || req.auth.role!=='teacher')
+        {
+            return res.status(403).json({error:'Only teachers can create activities.'})
+        }
+
+        const lastActivity = await activity.findOne({ order: [['code', 'DESC']] });
+        const nextCode = lastActivity ? lastActivity.code + 1 : 1;
```

Comment:
quick question, what's the purpose of `code`? is this some sort of `id` or rather an `access code` students can use to access the course?

Reply:
Exactly, I thought it would be a unique code that students can use to register for the course. I wanted this code to be generated automatically because it seemed simpler and that's why I used this increment.

Reply:
Awesome, sounds nice! I see that in order for you to compute this next code, you're doing a database transaction first, do you think we can implement this differently so that a database operation is not needed first? (database operations usually introduce latency, here it won't be necessarily felt because it's a small project, but usually we want to have **the smallest number of potentially latent operations**)

A couple options I can think of that may need some investigation:
- Investigate if **auto-increment** works in your database for fields which are not primary key, and let the database compute the new code upon inserting the activity
- Randomly generate a numeric code in the application without depending on previous rows stored in the database (either by hand, or maybe via an `npm` library)

Take this as an improvement point, if there's time remaining while building your backend, feel free to address this, otherwise, just keep this in mind for the future.

Really nice idea overall 😇